### DISCUSSION TITLE
Does strapi ignores foreign key constraints in database

### DISCUSSION THREAD

**Discussion by sanojsilva** [strapi/strapi#7694] [Legacy Topics]
Hey I have added some foreign key constraints to my postgres database. So if i try to delete a record i should get a foreign key violation error. But when i delete a record from the strapi admin panel it deletes that record without any errors. How do i stop that? Thank You!

**Comment by derrickmehaffy** [comment]
Strapi doesn't currently use foreign keys yet, it's on our list of things to implement.

**Reply by sanojsilva** [reply]
Yes I've got to know that. But my question is if I enforce those constraints manually directly in the database strapi can't delete that record right if it is referenced by some other table?

**Reply by derrickmehaffy** [reply]
I'm not sure in this case, I believe it would throw a DB error but not sure what that would look like to the user, most likely it would just throw a 500 error.

**Reply by sanojsilva** [reply]
Yes I've tried deleting manually through a sql query. Then it throws the foreign key violation error. But when I do through strapi panel it deletes the record without throwing any error :disappointed:

**Reply by derrickmehaffy** [reply]
:thinking: yeah for the moment we don't quite have that control yet but it's something we are looking at doing (along with proper migrations) Currently at the moment I don't have any ETA, you could possibly use the model lifecycles which work for both the Admin and user routes to check before you delete something.

At least a potential workaround

**Reply by sanojsilva** [reply]
Yes thank you for the idea I'll use lifecycles as a workaround.
